src/main/java/core/basesyntax/DataConverterImpl.java

```
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int LINE_LENGTH = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> dataFromFile) {
+        if (dataFromFile.isEmpty()) {
+            throw new RuntimeException("Empty list!");
+        }
+        return dataFromFile.stream()
+                .skip(1)
+                .map(line -> line.split(","))
+                .filter(lines -> lines.length == LINE_LENGTH)

```

This line filters out malformed lines (those not having exactly three fields). If the task requires strict validation (i.e., throw an exception on malformed input), you should throw an exception here instead of silently skipping. Please check the task requirements regarding error handling for malformed input.